src/code/z_collision_check.c

```diff
@@ -3086,90 +3407,105 @@ void func_80062B80(GlobalContext* globalCtx, Vec3f* v) {
     D_8015D3D8.speed = 8.0f;
     D_8015D3D8.gravity = -1.0f;
 
-    Effect_Add(globalCtx, &sp24, EFFECT_SPARK, 0, 1, &D_8015D3D8);
+    Effect_Add(globalCtx, &waterIndex, EFFECT_SPARK, 0, 1, &D_8015D3D8);
 }
 
-void func_80062CD4(GlobalContext* globalCtx, Vec3f* v) {
-    static EffectShieldParticleInit init = {
+/**
+ * Spawns streaks of light from hits against solid objects
+ */
+void CollisionCheck_SpawnShieldParticles(GlobalContext* globalCtx, Vec3f* v) {
+    static EffectShieldParticleInit sInitMetal = {
         16,
-        { 0.0f, 0.0f, 0.0f },
-        { 0x00, 0xC8, 0xFF, 0xFF },
-        { 0xFF, 0xFF, 0xFF, 0xFF },
-        { 0xFF, 0xFF, 0x80, 0xFF },
-        { 0xFF, 0xFF, 0x00, 0xFF },
-        { 0xFF, 0x40, 0x00, 0xC8 },
-        { 0xFF, 0x00, 0x00, 0xFF },
+        { 0, 0, 0 },
+        { 0, 200, 255, 255 },
+        { 255, 255, 255, 255 },
+        { 255, 255, 128, 255 },
+        { 255, 255, 0, 255 },
+        { 255, 64, 0, 200 },
+        { 255, 0, 0, 255 },
         2.1f,
         35.0f,
         30.0f,
         8,
-        { 0, 0, 0, 0x00, 0x80, 0xFF, 0x00, 300 },
-        1,
+        { 0, 0, 0, { 0, 128, 255 }, false, 300 },
+        true,
     };
-    s32 sp24;
+    s32 shield;
 
-    init.position.x = (s32)v->x;
-    init.position.y = (s32)v->y;
-    init.position.z = (s32)v->z;
-    init.lightPoint.x = init.position.x;
-    init.lightPoint.y = init.position.y;
-    init.lightPoint.z = init.position.z;
+    sInitMetal.position.x = v->x;
+    sInitMetal.position.y = v->y;
+    sInitMetal.position.z = v->z;
+    sInitMetal.lightPoint.x = sInitMetal.position.x;
+    sInitMetal.lightPoint.y = sInitMetal.position.y;
+    sInitMetal.lightPoint.z = sInitMetal.position.z;
 
-    Effect_Add(globalCtx, &sp24, EFFECT_SHIELD_PARTICLE, 0, 1, &init);
+    Effect_Add(globalCtx, &shield, EFFECT_SHIELD_PARTICLE, 0, 1, &sInitMetal);
 }
 
-void func_80062D60(GlobalContext* globalCtx, Vec3f* v) {
-    func_80062CD4(globalCtx, v);
+/**
+ * Spawns streaks of light and makes a metallic sound
+ */
+void CollisionCheck_ShieldParticlesMetal(GlobalContext* globalCtx, Vec3f* v) {
```

Comment:
It would probably be a good idea to always include "spawn" in the name, otherwise its purpose could be slightly confusing.